both/methods.js

```diff
@@ -153,16 +153,25 @@ Meteor.methods({
 
     Meteor.call('refreshRoundCodes', competitionId, eventCode);
   },
-  addNonEventRound: function(competitionId, round) {
+  addScheduleEvent: function(competitionId, eventData, roundId) {
     check(competitionId, String);
     throwIfCannotManageCompetition(this.userId, competitionId);
-    return Rounds.insert({
+
+    var round = Rounds.findOne(roundId);
+
+    return ScheduleEvents.insert({
       competitionId: competitionId,
-      title: round.title,
-      startMinutes: round.startMinutes,
-      durationMinutes: round.durationMinutes,
+      roundId: (round ? round._id : null),
```

Comment:
I thought about this for a while. Something has been nagging me about this, and I couldn't figure out what it was. The point about a stacktrace giving you line numbers is part of it, but is kind of a moot point when dealing with concatenated and minified javascript.

I think my problem is "print error and return" is that it doesn't allow us to refactor our code. Throwing exceptions lets us actually unwind the stack and stop executing code when something has gone wrong. Consider the following code.

```javascript
function bar() {
    if(....) {
        console.err("uh-oh! this shouldn't have happened");
        return;
    }
}
Meteor.methods({
    foo: function() {
        bar();
        functionThatShouldOnlyBeRunAfterBar();
    }
);
```